libraries/utils/utils/include/thread_pool.h

```diff
@@ -31,22 +31,25 @@
 #include <functional>
 #include <stdexcept>
 
-namespace bnb {
-    class thread_pool {
+namespace bnb
+{
+    class thread_pool
+    {
     public:
         thread_pool(size_t);
 
         template<class F, class... Args>
-        auto enqueue(F&& f, Args&&... args) 
+        auto enqueue(F&& f, Args&&... args)
             -> std::future<typename std::result_of<F(Args...)>::type>;
```

Comment:
better to use: -> std::future<typename std::invoke_result<F, Args...>::type>;

Reply:
Or better use thread_pool.hpp from OEP-module. Because code duplication:
https://github.com/Banuba/OEP-module/blob/master/offscreen_effect_player/thread_pool.h

Reply:
@dmitrystoyanovsky, what about this: "Or better use thread_pool.hpp from OEP-module."

Reply:
Good idea, will change.